File: src/main/java/com/IceCreamQAQ/Yu/event/EventListenerInfo.java
package com.IceCreamQAQ.Yu.event;

import com.IceCreamQAQ.Yu.annotation.Event;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.lang.reflect.Method;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class EventListenerInfo {

    private Class<?> clazz;
    private Method method;
    private Event.Weight weight;
    private EventInvoker invoker;
    private Object instance;

}

File: src/main/kotlin/com/IceCreamQAQ/Yu/event/EventListenerInfo.kt
package com.IceCreamQAQ.Yu.event

import com.IceCreamQAQ.Yu.annotation.Event.Weight
import com.IceCreamQAQ.Yu.fullName
import java.lang.reflect.Method

data class EventListenerInfo(
    var clazz: Class<*>?,
    val method: Method,
    val weight: Weight,
    val invoker: EventInvoker,
    val instance: Any? = null,
) {
    val methodFullName = method.fullName
}
